tools/assetpipelineeditor/exportdialog.cpp

```diff
@@ -0,0 +1,104 @@
+#include "exportdialog.h"
+#include "ui_exportdialog.h"
+
+#include <Kuesa/private/gltf2exporter_p.h>
+
+#include <QFile>
+#include <QFileInfo>
+#include <QDir>
+#include <QJsonDocument>
+#include <QFileDialog>
+#include <QStyle>
+#include <QPushButton>
+#include <QMessageBox>
+
+ExportDialog::ExportDialog(Kuesa::GLTF2Exporter &exporter, const QString &path, QWidget *parent)
+    : QDialog(parent)
+    , ui(new Ui::ExportDialog)
+    , m_exporter(exporter)
+    , m_originalFile(path)
+    , m_targetFile(m_originalFile)
+{
+    ui->setupUi(this);
+
+    ui->sourceLabel->setText(tr("Saving: ") + m_originalFile);
+
+    auto save_btn = ui->buttonBox->button(QDialogButtonBox::Save);
+    connect(save_btn, &QPushButton::clicked,
+            this, &ExportDialog::onSave);
+}
+
+ExportDialog::~ExportDialog()
+{
+    delete ui;
+}
+
+void ExportDialog::onSave()
+{
+    ui->errorLog->setPlainText("");
+
+    m_targetFile = QFileDialog::getSaveFileName(this, tr("Select a folder to save to"), m_targetFile, tr("glTF 2.0 file (*.gltf)"));
+
+    const QDir orig_dir = QFileInfo(m_originalFile).dir();
+    const QDir target_dir = QFileInfo(m_targetFile).dir();
+    if (orig_dir.absolutePath() == target_dir.absolutePath()) {
+        const int res = QMessageBox::warning(
+                this, tr("Destructive operation"),
+                tr("Saving in the original folder will overwrite assets."),
+                QMessageBox::Ok | QMessageBox::Cancel,
+                QMessageBox::Cancel);
+        if (res != QMessageBox::Ok) {
+            ui->errorLog->setPlainText(tr("Nothing saved."));
+            return;
```

Comment:
Unsure since I couldn't try it yet (I lack draco), but isn't it going to close the dialog still when getting there? If yes this should be fixed.

Reply:
hmm, on my machine (linux) the export dialog isn't closed after saving, but maybe this is something platform-specific - I will try on mac & windows

Reply:
OK, just me being confused on save accepting the dialog or not. I was surely confused, feel free to ignore this.